assignments/prototype-refactor.js

```diff
@@ -1,9 +1,224 @@
+
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properties and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+/*
+  === CharacterStats ===
+  * healthPoints
+  * name
+  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+  * should inherit destroy() from GameObject's prototype
+*/
 
+/*
+  === Humanoid (Having an appearance or character resembling that of a human.) ===
+  * team
+  * weapons
+  * language
+  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+  * should inherit destroy() from GameObject through CharacterStats
+  * should inherit takeDamage() from CharacterStats
+*/
+ 
+/*
+  * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+  * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+  * Instances of CharacterStats should have all of the same properties as GameObject.
 */
+
+class GameObject {
+    constructor(attributes) {
+        this.createdAt = attributes.createdAt;
+        this.dimensions = attributes.dimensions;
+    }
+    destroy() {
+        return `${this.name} was removed from the game.`;
+    }
+}
+
+class CharacterStats extends GameObject {
+    constructor(characterAttributes) {
+        super(characterAttributes);
+        this.healthPoints = characterAttributes.healthPoints;
+        this.name = characterAttributes.name;
+    }
+    takeDamage() {
+        return `${this.name} took damage.`;
+    }
+}
+
+class Humanoid extends CharacterStats {
+    constructor(hAttributes) {
+        super(hAttributes);
+        this.team = hAttributes.team;
+        this.weapons = hAttributes.weapons;
+        this.language = hAttributes.language;
+    }
+    greet() {
+        return `${this.name} offers a greeting in ${this.language}`;
+    }
+}
+
+class Hero extends Humanoid {
+    constructor(heroAttributes) {
+        super(heroAttributes);
+        this.isHero = heroAttributes.isHero;
+        this.nickname = heroAttributes.nickname;
+    }
+    removeHealthPoints() {
+        let newHealthpoint = this.healthPoints--;
+        return newHealthpoint;
+    }
+    sayRole() {
+        if(this.isHero) {
+            return `I'm a hero`;
+          } else {
+            return `I'm not a hero`;
+          }
+    }
+}
+
+class Villain extends Humanoid {
+    constructor(villainAttributes) {
```

Comment:
Great job going the extra mile and converting the stretch portion so it still works.